department_names.py

```diff
@@ -4,61 +4,86 @@
 import logging
 import os
 import requests
-import yaml
+import csv
 
-portal_url = os.environ.get('PORTAL_URL', "https://vulekamali.gov.za/")
+portal_url = os.environ.get("PORTAL_URL", "https://vulekamali.gov.za/")
 
 department_names = {
-    'national': {},
-    'provincial': {},
+    "2014-15": {"national": {}, "provincial": {},},
+    "2015-16": {"national": {}, "provincial": {},},
+    "2016-17": {"national": {}, "provincial": {},},
+    "2017-18": {"national": {}, "provincial": {},},
+    "2018-19": {"national": {}, "provincial": {},},
+    "2019-20": {"national": {}, "provincial": {},},
 }
+
 warned = {}
 
 
 def modify_datapackage(datapackage, parameters, stats):
     # We're not modifying the datapackage but we execute here to execute once
     # before processing rows.
-    year_slug = parameters['financial_year']
-    sphere = parameters['sphere']
-    listing_url_path = year_slug + '/departments.yaml'
-    listing_url = portal_url + listing_url_path
-    r = requests.get(listing_url)
-    r.raise_for_status()
-    response = yaml.load(r.text)
-    for government in response[sphere]:
-        department_names[sphere][government['name']] = {}
-        for department in government['departments']:
-            department_names[sphere][government['name']][department['slug']] \
-                = department['name']
+    financial_years = list(department_names.keys())
+    sphere = parameters["sphere"]
+    for fin_year in financial_years:
+        governments = []
+        listing_url_path = fin_year + "/departments.csv"
+        listing_url = portal_url + listing_url_path
+        r = requests.get(listing_url)
```

Comment:
Please raise an exception if error code was not 200 or 404

Reply:
If it raises an exception, it doesn't create a department_names dictionary and it breaks the pipeline. There was r.raise_for_status() before and I changed it because of that

Reply:
Solved

https://github.com/vulekamali/treasury-pipelines/pull/33/commits/c48e4d5feadd50cac386f13e4c3eadabec6d46fd